ccmlib/scylla_node.py

```diff
@@ -1041,19 +1041,24 @@ def run_patchelf(patchelf_cmd):
                                    os.path.join(self.get_bin_dir(), 'scylla-jmx-1.0.jar'), replace=replace)
             self.hard_link_or_copy(os.path.join(self.get_jmx_dir(), 'scylla-jmx'),
                                    os.path.join(self.get_bin_dir(), 'scylla-jmx'), replace=replace)
+            select_java = Path(self.get_jmx_dir()) / 'select-java'
         else:
             self.hard_link_or_copy(os.path.join(self.get_jmx_dir(), 'target', 'scylla-jmx-1.0.jar'),
                                    os.path.join(self.get_bin_dir(), 'scylla-jmx-1.0.jar'), replace=replace)
             self.hard_link_or_copy(os.path.join(self.get_jmx_dir(), 'scripts', 'scylla-jmx'),
                                    os.path.join(self.get_bin_dir(), 'scylla-jmx'), replace=replace)
+            select_java = Path(self.get_jmx_dir()) / 'scripts' / 'select-java'
 
         os.makedirs(os.path.join(self.get_bin_dir(), 'symlinks'), exist_ok=exist_ok)
         scylla_jmx_file = os.path.join(self.get_bin_dir(), 'symlinks', 'scylla-jmx')
         if os.path.exists(scylla_jmx_file) and replace:
             os.remove(scylla_jmx_file)
-        java_home = os.environ.get('JAVA_HOME', '/usr')
-        java_exe = os.path.join(java_home, 'bin', 'java')
-        os.symlink(java_exe, scylla_jmx_file)
+        if select_java and not os.environ.get('JAVA_HOME'):
```

Comment:
I've decided that if the use want to point to a specific JVM we'll still gonna honor JAVA_HOME

Reply:
no need to check `select_java` here, as it is always evaluated to True here. as  you assigns to this variable with a `Path` in both branches above. i don't see another branch leading to this line.

also, can we favor positive conditions over negative ones? i'd suggest use something like:
```py
java_home = os.environ.get('JAVA_HOME')
if java_home:
    java_exe = os.path.join(java_home, 'bin', 'java')
    os.symlink(java_exe, scylla_jmx_file)
else:
    # use select-java as a fallback, if $JAVA_HOME is not defined.
    os.symlink(select_java, scylla_jmx_file)
```

Reply:
> no need to check `select_java` here, as it is always evaluated to True here. as  you assign to this variable with a `Path` in both branches above. i don't see another branch leading to this line.
> 
> also, can we favor positive conditions over negative ones? i'd suggest use something like:
> ```py
> java_home = os.environ.get('JAVA_HOME')
> if java_home:
>     java_exe = os.path.join(java_home, 'bin', 'java')
>     os.symlink(java_exe, scylla_jmx_file)
> else:
>     # use select-java as a fallback, if $JAVA_HOME is not defined.
>     os.symlink(select_java, scylla_jmx_file)
> ```

I can reverse the logic, but we should still support case select-java doesn't exist, i.e. old releases

Reply:
Anyway I can confirm the current code works if you do have the select-java script (new Scylla). Thanks.

Reply:
fix the logic to be a bit more straight forward, with no negatives